### PR TITLE
refactor: add minor code clean-up

### PR DESCRIPTION
**Pull request by tranzystorekk**
Highlights:

- Add `Deref` and `IntoIterator` impls for `SmallVec` references for convenience
- Shorten code where deref coercion can be applied
- Replace `&*` with explicit `.deref()` calls for readability

### CODE REVIEW

**Comment by Eh2406** [comment]
Thanks for the PR! 

Just a few questions:
I like the use of `Deref` on `SmallVec`! Can we double down on that by inlining and removing `as_slice`?
I would have expected `IntoIterator` to have `Item = T;`. Now that we have `Deref`, can we just call `smallvec.iter()` to get an iterator of `&T`? That seems less surprising; if a little more verbose.
Can you elaborate on why you fined `.deref()` better to read then `&*`? Not that I always like Clippys opinions, but it has a [(pedantic) lint](https://rust-lang.github.io/rust-clippy/master/#explicit_deref_methods) to go the other way. 

Over all I like the direction!

**Comment by tranzystorekk** [comment]
> Can you elaborate on why you fined `.deref()` better to read then `&*`? Not that I always like Clippys opinions, but it has a [(pedantic) lint](https://rust-lang.github.io/rust-clippy/master/#explicit_deref_methods) to go the other way.

Huh, I didn't know clippy had that lint, but I guess that's why it's pedantic 😄 I know it's a matter of opinion, but I think the deref coercion, although slick, is a very confusing and not well understood mechanism. The verbose `.deref()` call clearly says what is done when we use it in the `match` expressions.

> Can we double down on that by inlining and removing as_slice?

We can remove the `SmallVec::iter()` method and delegate to `slice::iter()` with deref, but `IntoIterator` is mostly for all those places where I replaced `for x in smallvec.iter() { ... }` with `for x in &smallvec { ... }`. Do you think those should be reverted to the verbose version?

Also, `SmallVec::as_slice()` cannot be removed, because deref coercion doesn't work in `match` expressions, i.e. all those places where we do

```rs
match smallvec.as_slice() {
    [x] => ...
    ...
}
```

EDIT: I guess we could call `.deref()` in such `match`es, but in this case I think `as_slice()` conveys what we want to match against more clearly.

**Review by Eh2406 — APPROVED** [review]
`as_slice` and `deref` both have context where they are clearer. I am convinced, have them both.
As to `IntoIterator`, I miss read your PR. It is on `&'a SmallVec<T>`, so `Item = &'a T` makes sense. All good, I like it now.
For `.deref()` vs `&*`, I don't have a strong opinion. Ether way is fine. I will leave it to @mpizenberg.

So sorry for doing to fast a review earlier. I think this is good.

Our normal prosses on this project is that before we merge we give 10 days for less active project members to put in a comment. So please be a little patient.

File: src/range.rs
Comment (mpizenberg) [inline src/range.rs]: The destructuring of `(start, end)` is nice but for the body of the function, I'd prefer if we keep a match. We can match on just `end` though.

File: src/report.rs
Comment (mpizenberg) [inline src/report.rs]: I'd prefer if we keep two lines here to keep the style consistent with the rest of the crate.
```rust
use std::fmt;
use std::ops::{Deref, DerefMut};
```

**Review by mpizenberg — CHANGES_REQUESTED** [review]
Thanks! I don't have any strong opinion on the deref style, just as long as we keep clippy happy :) I generally like the changes! Have a few nitpicks though if you can address them.

Also don't hesitate to simply push new commits as it's easier to review after an initial review instead of force push. We squash merge anyway so there will be only one commit left.

File: src/range.rs
Comment (tranzystorekk) [reply src/range.rs]: Done ✅

File: src/report.rs
Comment (tranzystorekk) [reply src/report.rs]: Done ✅

**Review by mpizenberg — APPROVED** [review]

**Review by mpizenberg — APPROVED** [review]

**Comment by Eh2406** [comment]
Thank you for the clean up!

**Pull request merged**
